### PR TITLE
[WIP] Replace lib/xapi/* with trackler gem

### LINKED ISSUE #141: Refactor APIs to use the Trackler gem
Right now the exercism.io is jumping through loops to call the x-api endpoints in order to display track and exercise data. This is pretty slow, and it's also a bit of a mess.

We've extracted the core track and exercise logic into a new gem, [trackler](http://github.com/exercism/trackler), so that we can use the same code in both exercism.io and x-api efficiently.

The TL;DR is that we need to swap out calls to `Xapi` collections to use the equivalent `Trackler` collections.

* `Xapi.problems` => `Trackler.problems`
* `Xapi.tracks` => `Trackler.tracks`
* `Xapi.implementations` => `Xapi.implementations`

There are a very small number of API-specific things inside of lib/xapi*, so we may need to make some small tweaks in order to make that happen.

If you want to help with this, then comment below with which file you're tackling, and submit your pull request to the #142 pull request by using the`trackler` base branch instead of to `master`. (If you have questions, ask!)

## TODO

For each step we need to

- swap out the collection
- run all of the tests to make sure that it works
- run the website with the `rackup` command
- visit the actual routes in the file to make sure that it works with the real track data

These are the files:

- [x] `api/helpers/finders.rb`
- [x] `api/v1/routes/docs.rb`
- [x] `api/v1/routes/exercises.rb`
- [x] `api/v1/routes/home.rb`
- [x] `api/v1/routes/problems.rb`
- [x] `api/v1/routes/tracks.rb`
- [x] `api/v3/routes/problems.rb`
- [x] `api/v3/routes/tracks.rb`

Then, once all of the routes are done, we can

- [x] delete `lib/xapi/*`

### PR DESCRIPTION
**Pull request by kytrinyx**
**TL;DR**: we need to swap out calls to `Xapi` collections to use the equivalent `Trackler` collections.

See the TODO list in #141 for details.

We've extracted all of the core x-api logic to the [trackler](http://github.com/exercism/trackler).

Closes #141

### CODE REVIEW

**Comment by kytrinyx** [comment]
Not sure why this is failing. My gem is here: https://rubygems.org/gems/trackler/versions/1.0.0.0

**Comment by kytrinyx** [comment]
OK, in order to move forward with this, I'm going to use the gem from GitHub instead of from RubyGems.
